maths/rkf45.py

```diff
@@ -0,0 +1,86 @@
+from collections.abc import Callable
+import numpy as np
+
+
+class RangeError(Exception):
+    "Will be raised when initial x is greater than or equal to final x"
+
+
+def rkf45(
+    ode: Callable,
+    y0: float,
+    x0: float,
+    step_size: float,
+    xn: float,
+) -> np.ndarray:
+    """
+    Solve ODE using Runge-Kutta-Fehlberg Method (rkf45) of order 5.
+
+    Reference: https://en.wikipedia.org/wiki/Runge%E2%80%93Kutta%E2%80%93Fehlberg_method
+
+    args:
+    ode (callable): Ordinary Differential Equation as function of x and y.
+    y0 (float) : Initial value of y.
+    x0 (float) : Initial value of x.
+    step_size (float) : Increament value of x (step-size).
+    xn (float) : Final value of x.
+
+    Returns:
+        np.ndarray: Solution of y at each nodal point
```

Comment:
Please do not repeat the data types that are checked by mypy in the function signature because the reader will be confused if one changes and the other does not.
Put x before y.
```suggestion
    func: Callable,
    x_initial: float,
    y_initial: float,
    step_size: float,
    x_final: float,
) -> np.ndarray:
    """
    Solve an Ordinary Differential Equation (ODE) function using
    Runge-Kutta-Fehlberg Method (rkf45) of order 5.

    Reference: https://en.wikipedia.org/wiki/Runge%E2%80%93Kutta%E2%80%93Fehlberg_method

    args:
    func: Ordinary Differential Equation as a function of x and y.
    x_initial: Initial value of x.
    y_initial: Initial value of y.
    step_size: Increament value of x (step-size).
    x_final: Final value of x.

    Returns:
        Solution of y at each nodal point
```